wp-content/themes/core/blocks/tribe/terms/render.php

```diff
@@ -0,0 +1,85 @@
+<?php declare(strict_types=1);
+
+// get block settings (attributes)
+$taxonomy         = $attributes['taxonomyToUse'] ?? 'category';
+$only_primay_term = $attributes['onlyPrimaryTerm'] ?? false;
+$has_links        = $attributes['hasLinks'] ?? false;
+$style            = $attributes['termStyle'] ?? 'default';
+$terms            = false;
+$primary_term     = false;
+
+/**
+ * If we should only display the primary term:
+ * - If we have yoast we can get the primary term that way
+ * - Otherwise, we'll have to grab the first term in the list
+ *
+ * If we should display all terms:
+ * - Grab all terms for the taxonomy
+ */
+if ( $taxonomy === 'category' && $only_primay_term && class_exists( 'WPSEO_Primary_Term' ) ) {
+	$wpseo_primary_term = new WPSEO_Primary_Term( $taxonomy, get_the_ID() );
+	$wpseo_primary_term = $wpseo_primary_term->get_primary_term();
+	if ( $wpseo_primary_term ) {
+		$primary_term = get_term( $wpseo_primary_term );
+	} else {
+		$terms        = get_the_terms( get_the_ID(), $taxonomy );
+		$primary_term = $terms ? $terms[0] : false;
+	}
+} elseif ( $only_primay_term ) {
+	$terms        = get_the_terms( get_the_ID(), $taxonomy );
+	$primary_term = $terms ? $terms[0] : false;
+} else {
+	$terms = get_the_terms( get_the_ID(), $taxonomy );
+}
+
+if ( ! $terms && ! $primary_term && ! is_admin() ) {
+	return;
+}
+?>
+
+<div <?php echo get_block_wrapper_attributes(); ?>>
```

Comment:
I'd liek to take a stab at cleaning up the PHP a bit here. It's pretty intense. 😬

Reply:
I didn't want to make everything a list if it didn't need to be a list, or else I suppose I could remove that whole first block. Is there anything specific that isn't clear about how things display? Would it be helpful to move things out of `sprintf` calls maybe?

Reply:
I think this php has been made a little overly complex here.  if you don't mind, @GeoffDusome, I am going to take a  stab at a PR to clean some of this up.